the-blue-alliance-ios/View Elements/Match/MatchSummaryView.swift

```diff
@@ -144,30 +156,67 @@ class MatchSummaryView: UIView {
     private func teamLabel(for teamKey: String, baseTeamKey: String?, dq: Bool) -> UILabel {
         let text: String = "\(Team.trimFRCPrefix(teamKey))"
         let isBold: Bool = (teamKey == baseTeamKey)
-
+        
         return label(text: text, isBold: isBold, isStrikethrough: dq)
     }
+    
+    private func teamButton(for teamKey: String, baseTeamKey: String?, dq: Bool) -> UIButton {
+        let text: String = "\(Team.trimFRCPrefix(teamKey))"
+        let isBold: Bool = (teamKey == baseTeamKey)
+
+        return button(text: text, isBold: isBold, isStrikethrough: dq)
+    }
+    
+    private func button(text: String, isBold: Bool, isStrikethrough: Bool = false, isTeam: Bool = true) -> UIButton {
+        let newButton = UIButton(type: .system)
+        newButton.setTitle(text, for: [])
+        
+        // set color of the label to black like default
+        let buttonColor = UIColor.primaryBlue
+        newButton.setTitleColor(buttonColor, for: .normal)
+        // change color to gray when user hits the button
+        newButton.setTitleColor(buttonColor.withAlphaComponent(0.8), for: .highlighted)
+        newButton.setTitleColor(buttonColor.withAlphaComponent(0.8), for: .focused)
+        
+
+        newButton.titleLabel?.attributedText = customAttributedString(text: text, isBold: isBold, isStrikethrough: isStrikethrough)
+        
+        newButton.addTarget(self, action: #selector(teamPressed(sender:)), for: .touchUpInside)
+        newButton.translatesAutoresizingMaskIntoConstraints = false
+        return newButton
+    }
 
     private func label(text: String, isBold: Bool, isStrikethrough: Bool = false) -> UILabel {
+        let label = UILabel()
+        label.textAlignment = .center
+        label.attributedText = customAttributedString(text: text, isBold: isBold, isStrikethrough: isStrikethrough)
+        label.translatesAutoresizingMaskIntoConstraints = false
+
+        return label
+    }
+    
+    @objc private func teamPressed(sender: UIButton) {
+        guard let text = sender.titleLabel?.text else { return }
```

Comment:
Not *exactly* the same (since we're indexing in to lists, as opposed to fetching teams) but here's some examples of what I'm talking about, where we'll set the tag on a view and use it later to get some data

https://github.com/the-blue-alliance/the-blue-alliance-ios/blob/master/the-blue-alliance-ios/View%20Elements/Events/EventAllianceTableViewCell.swift#L91 

https://github.com/the-blue-alliance/the-blue-alliance-ios/blob/master/the-blue-alliance-ios/View%20Elements/Awards/AwardTableViewCell.swift#L65

Reply:
Tags would be better but we could also create an extensible tappable UIButton like you said that contains TeamKeys themselves which would also solve this issue.